crates/storage/src/collections/binary_heap/children_vector.rs

```diff
@@ -0,0 +1,460 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides an interface around the vector used to store elements of the
+//! [`BinaryHeap`](`super::BinaryHeap`) in storage. This is necessary since
+//! we don't just store each element in it's own storage cell, but rather
+//! optimize storage access by putting children together in one storage cell.
+
+use super::{
+    children,
+    children::Children,
+    StorageVec,
+};
+use crate::{
+    collections::extend_lifetime,
+    traits::{
+        KeyPtr,
+        PackedLayout,
+        SpreadLayout,
+    },
+    Lazy,
+};
+
+/// Provides an interface for accessing elements in the `BinaryHeap`.
+///
+/// Elements of the heap are stored in a vector of `Children` objects, whereby
+/// each `Children` object contains two elements. When operating on indices of
+/// the `BinaryHeap` this interface transposes heap indices to the child inside
+/// the `Children` object, in which the element is stored.
+#[derive(Default, PartialEq, Eq, Debug)]
+pub struct ChildrenVector<T>
```

Comment:
```suggestion
pub struct ChildrenVec<T>
```

I would personally prefer that since `Vector` is almost always abbreviated to `Vec`